src/codegen/block.rs

```diff
@@ -194,20 +197,31 @@ pub fn compile_block<'a, 'b>(
                 b.build_cond_br(v, if_branch_block, Some(else_branch_block));
 
                 // Then compile what's inside the if statement. This will take us through the else to the end statement
+                let else_built = Cell::new(false);
                 remaining_instructions = compile_block(
                     m_ctx,
                     f_ctx,
                     breakout_stack,
                     &inner_termination_target,
                     Some(IfCtx {
                         else_block: else_branch_block,
-                        else_locals: else_branch_locals,
+                        else_locals: else_branch_locals.clone(),
+                        else_built: &else_built
                     }),
                     if_branch_locals,
                     if_branch_block,
                     remaining_instructions,
                 );
 
+                // Handle the case where there is no else
+                if !else_built.get() {
+                    let fake_stack = Vec::new(); // Tacitly assumes an if with no else can't produce a value
```

Comment:
@bushidocodes Just wanted to confirm that this assumption is correct, and you just learned a bunch of wasm ;-) Does this assumption make sense? Also, if @Others doesn't need the sanity check here, I'm happy with it.

Reply:
[Re-posting from Slack]. `if` expects and consumes a single argument/operand of type i32 which is evaluated as a bool (0 is false. everything else is true). This is logically popped from the stack and consumed. All conditional logic is terminated by either an `else` or an `end` , both of which expect no arguments/operands. That means that the instructions in the unit of execution following an if or an else can add to the stack, but when this unit of execution terminates (as indicated by an `else` or an `end`), the stack must be in the same state as when it started. So the net impact of running a complex if/else is that a single value evaluated as a boolean for the conditional is popped from the stack. Stated differently, I think that means that "an if of any kind cannot produce a value"